src/unix/fs.c

```diff
@@ -404,6 +404,36 @@ static ssize_t uv__fs_readlink(uv_fs_t* req) {
   return 0;
 }
 
+static ssize_t uv__fs_realpath(uv_fs_t* req) {
+  ssize_t len;
+  char* buf;
+
+  len = pathconf(req->path, _PC_PATH_MAX);
+
+  if (len == -1) {
+#if defined(PATH_MAX)
+    len = PATH_MAX;
+#else
+    len = 4096;
+#endif
+  }
+
+  buf = uv__malloc(len + 1);
+
+  if (buf == NULL) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  if (realpath(req->path, buf) == NULL) {
```

Comment:
*A bit late note:*

@bnoordhuis 
> POSIX pedantry: If resolved_name is a null pointer, the behavior of realpath() is implementation-defined.

That was correct for older POSIX versions only, in the current version passing a NULL pointer as the `resolved_name` is property documented: http://pubs.opengroup.org/onlinepubs/9699919799/functions/realpath.html